File: Java/app/src/main/java/com/flyingbee/FPPDFConverterDemoJava/DemoApplication.java
package com.flyingbee.FPPDFConverterDemoJava;

import android.app.Application;

/**
 * Boots the process-wide {@link ConverterController} singleton so every screen
 * shares the same state (the Java counterpart of the Kotlin demo's shared
 * AndroidViewModel).
 */
public final class DemoApplication extends Application {
    @Override
    public void onCreate() {
        super.onCreate();
        ConverterController.init(this);
    }
}

File: Kotlin/app/src/main/kotlin/com/flyingbee/FPPDFConverterDemo/DemoApplication.kt
package com.flyingbee.FPPDFConverterDemo

import android.app.Application

// ---------------------------------------------------------------------------
// DemoApplication : creates the process-wide ConverterController singleton in
// onCreate so every Activity (Home + the 8 sub-screens) shares the same state,
// mirroring the Java demo's DemoApplication.
// ---------------------------------------------------------------------------

class DemoApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        ConverterController.init(this)
    }
}
